substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -338,9 +338,10 @@ impl<T: Config> Pallet<T> {
 
 		let dest = Self::payee(StakingAccount::Stash(stash.clone()));
 		let maybe_imbalance = match dest {
-			RewardDestination::Controller => Self::bonded(stash)
-				.map(|controller| T::Currency::deposit_creating(&controller, amount)),
-			RewardDestination::Stash => T::Currency::deposit_into_existing(stash, amount).ok(),
+			// Note: Controller accounts are being deprecated and will no longer point to a separate
+			// account. `Controller` variant now points to the `stash`.
+			RewardDestination::Stash | RewardDestination::DeprecatedController =>
+				T::Currency::deposit_into_existing(stash, amount).ok(),
```

Comment:
In addition to this note, you'd have an `#[allow(deprecated)]` here.

Reply:
Oh that's great, thanks for the tips man! I'll push those updates momentarily.